packages/vega-parser/src/parsers/guides/axis-title.js

```diff
@@ -58,13 +77,45 @@ export default function(spec, config, userEncode, dataRef) {
     align:       _('titleAlign')
   });
 
-  !addEncode(encode, 'x', _('titleX'), 'update')
-    && !horizontal && !has('x', userEncode)
-    && (encode.enter.auto = {value: true});
+  if (isSignal(orient)) {
+    if (_('titleX') != null) {
+      delete encode.update['x'][0].signal;
+      encode.update['x'][0] = {
+        ...encode.update['x'][0],
+        ..._('titleX')
+      };
+    } else {
+      if (!has('x', userEncode)) {
+        encode.enter.auto = xyAxisConditionalEncoding('y', orient.signal, { value: true }, null);
+      }
+    }
+
+    if (_('titleY') != null) {
+      delete encode.update['y'][0].signal;
+      encode.update['y'][0] = {
+        ...encode.update['y'][0],
```

Comment:
[Spread for object literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals) is a more recent addition to some browsers. I've been avoiding it in Vega so that we don't run into backwards compatibility issues. (Even though I love them...)

Reply:
Maybe we should add babel even for our normal builds. I can take a stab at a pull request at some point if you are open to accepting it.

Reply:
I've switched these out for `extend()` for now.